src/s390_crypto.c

```diff
@@ -318,14 +325,114 @@ static void set_switches(int msa)
 			*s390_kdsa_functions[n].enabled = 1;
 }
 
+unsigned int is_device_online(const char *dev)
+{
+	unsigned int ret = 0;
+	FILE *file;
+	char c;
+
+	if ((file = fopen(dev, "r")) == NULL)
+		return 0;
+
+	/* Check if device online: 0 = offline, 1 = online */
+	if ((c = fgetc(file)) == '1')
+		ret = 1;
+
+	fclose(file);
+
+	return ret;
+}
+
+unsigned int get_device_type(const char *dev, char *devtype)
+{
+	unsigned int ret = 0;
+	char *type = NULL;
+	size_t size = 0;
+	FILE *file;
+
+	file = fopen(dev, "r");
+	if (file == NULL)
+		return 0;
+
+	/* Read device type: e.g. 'CEX5C' */
+	if (getline(&type, &size, file) == -1)
+		goto end;
+
+	/* Did we get something like 'CEX5C\n'? */
+	if (strlen(type) != 6 || strstr(type, "CEX") != &type[0])
+		goto end;
+
+	memcpy(devtype, type, 5);
+	ret = 1;
+
+end:
+	free(type);
+	fclose(file);
+
+	return ret;
+}
+
+#define AP_PATH  "/sys/devices/ap"
+#define MAX_DEV_LEN 280
+
+unsigned int search_for_cards()
+{
+	DIR *sysDir;
+	unsigned int ret = 0;
+	char dev[MAX_DEV_LEN] = AP_PATH;
```

Comment:
Why the = AP_PATH ?
You use dev later with snprintf(dev, ...) so the assignment is not needed.

Reply:
Yes, but there's this stmt below which needs the path in dev:

> 	if ((sysDir = opendir(dev)) == NULL)
> 		return 0;